cpp/src/io/json/writer_impl.cu

```diff
@@ -0,0 +1,782 @@
+      auto const col_idx      = thrust::distance(d_table.begin(), itr);
+      auto const d_column     = *itr;
+      bool const null_element = d_column.is_null(idx);
+
+      if (write_separator && (include_nulls == true || !null_element)) {
+        if (d_buffer) d_buffer = strings::detail::copy_string(d_buffer, d_val_separator);
+        bytes += d_val_separator.size_bytes();
+        write_separator = false;
+      }
+
+      // column_name:
+      if ((include_nulls == true || !null_element) && !d_column_names.is_null(col_idx)) {
+        auto const d_name = d_column_names.element<string_view>(col_idx);
+        if (d_buffer) d_buffer = strings::detail::copy_string(d_buffer, d_name);
+        bytes += d_name.size_bytes();
+        if (d_buffer) d_buffer = strings::detail::copy_string(d_buffer, d_col_separator);
+        bytes += d_col_separator.size_bytes();
+      }
+
+      // write out column's row data (or narep if the row is null)
+      if (include_nulls == true || !null_element) {
+        auto const d_str = null_element ? d_narep.value() : d_column.element<string_view>(idx);
+        if (d_buffer) d_buffer = strings::detail::copy_string(d_buffer, d_str);
+        bytes += d_str.size_bytes();
+      }
+
+      write_separator = write_separator || (include_nulls == true) || !null_element;
+    }
+    if (d_buffer) d_buffer = strings::detail::copy_string(d_buffer, append_row);
+    bytes += append_row.size_bytes();
+
+    if (!d_chars) d_offsets[idx] = bytes;
+  }
+};
+
+/**
+ * @brief Concatenate the strings from each row of the given table as structs in JSON string
+ *
+ * Each row will be struct with field name as column names and values from each column in the table.
+ *
+ * @param strings_columns Table of strings columns
+ * @param column_names Column of names for each column in the table
+ * @param prepend_row Prepend this string to each row
+ * @param append_row  Append this string to each row
+ * @param column_name_separator  Separator between column name and value
+ * @param value_separator Separator between values
+ * @param narep Null-String replacement
+ * @param include_nulls Include null string entries in the output
+ * @param stream CUDA stream used for device memory operations and kernel launches.
+ * @param mr Device memory resource to use for device memory allocation.
+ * @return New strings column of JSON structs in each row
+ */
+std::unique_ptr<column> struct_to_strings(table_view const& strings_columns,
+                                          column_view const& column_names,
+                                          string_scalar const& prepend_row,
+                                          string_scalar const& append_row,
+                                          string_scalar const& column_name_separator,
+                                          string_scalar const& value_separator,
+                                          string_scalar const& narep,
```

Comment:
aLooks like these parameters should be (cudf::)`string_view`s, similar to host/device_span parameters.
Not sure if the d_ prefix (as used for local variable) is needed, since the type already reveals where the data is.

Reply:
updated to `string_view`